From e8996bf6d222cb8df448cece9d329bd9e863a72d Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Tue, 25 Sep 2018 19:51:13 +0300
Subject: [PATCH 23/24] ucm-imx8: Add initial board support

Add initial board support.
Add an option to select one of CompuLab's iMX8 boards.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm/cpu/armv8/imx8m/Kconfig |  6 +--
 arch/arm/dts/Makefile            |  1 +
 arch/arm/dts/ucm-imx8.dts        | 20 ++++++++++
 board/compulab/imx8/Kconfig      | 80 ++++++++++++++++++++++++++++++++++++++++
 configs/ucm-imx8_defconfig       | 36 ++++++++++++++++++
 5 files changed, 140 insertions(+), 3 deletions(-)
 create mode 100644 arch/arm/dts/ucm-imx8.dts
 create mode 100644 board/compulab/imx8/Kconfig
 create mode 100644 configs/ucm-imx8_defconfig

diff --git a/arch/arm/cpu/armv8/imx8m/Kconfig b/arch/arm/cpu/armv8/imx8m/Kconfig
index 2601d8f..1461d90 100644
--- a/arch/arm/cpu/armv8/imx8m/Kconfig
+++ b/arch/arm/cpu/armv8/imx8m/Kconfig
@@ -22,8 +22,8 @@ config TARGET_IMX8MQ_DDR4_ARM2
 	select IMX8M
 	select SUPPORT_SPL
 
-config TARGET_CL_SOM_IMX8
-	bool "cl-som-imx8"
+config TARGET_COMPULAB_IMX8
+	bool "compulab-imx8"
 	select IMX8M
 	select SUPPORT_SPL
 
@@ -34,6 +34,6 @@ config SYS_SOC
 
 source "board/freescale/imx8mq_evk/Kconfig"
 source "board/freescale/imx8mq_arm2/Kconfig"
-source "board/compulab/cl-som-imx8/Kconfig"
+source "board/compulab/imx8/Kconfig"
 
 endif
diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index b368856..77a3fe2 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -391,6 +391,7 @@ dtb-$(CONFIG_ARCH_IMX8M) += fsl-imx8mq-evk.dtb \
 		fsl-imx8mq-ddr4-arm2.dtb
 
 dtb-$(CONFIG_TARGET_CL_SOM_IMX8) += cl-som-imx8.dtb
+dtb-$(CONFIG_TARGET_UCM_IMX8) += ucm-imx8.dtb
 
 dtb-$(CONFIG_ARCH_IMX8) += fsl-imx8qm-lpddr4-arm2.dtb \
 		fsl-imx8qm-ddr4-arm2.dtb \
diff --git a/arch/arm/dts/ucm-imx8.dts b/arch/arm/dts/ucm-imx8.dts
new file mode 100644
index 0000000..53144b7
--- /dev/null
+++ b/arch/arm/dts/ucm-imx8.dts
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2017 CopuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "cl-som-imx8.dts"
+
+/ {
+	model = "CompuLab UCM-iMX8";
+	compatible = "compulab,ucm-imx8", "fsl,imx8mq";
+};
diff --git a/board/compulab/imx8/Kconfig b/board/compulab/imx8/Kconfig
new file mode 100644
index 0000000..3be6683
--- /dev/null
+++ b/board/compulab/imx8/Kconfig
@@ -0,0 +1,80 @@
+if TARGET_COMPULAB_IMX8
+
+choice
+    prompt "CompuLab iMX8 board select"
+
+config TARGET_CL_SOM_IMX8
+    bool "cl-som-imx8"
+
+config TARGET_UCM_IMX8
+    bool "ucm-imx8"
+
+endchoice
+
+config SYS_BOARD
+	default "cl-som-imx8"
+
+config SYS_VENDOR
+	default "compulab"
+
+config SYS_CONFIG_NAME
+	default "cl-som-imx8"
+
+choice
+	prompt "RAM Volume select"
+	default RAM_2_1G
+	help
+	 This option configures 2G RAM only
+
+config RAM_1G
+	bool "RAM_1G"
+	select RAM_SET_1G
+	help
+	 This option configures 1G RAM only
+
+config RAM_2G
+	bool "RAM_2G"
+	select RAM_SET_2G
+	help
+	 This option configures 2G RAM only
+
+config RAM_2_1G
+	bool "RAM_2_1G"
+	select RAM_SET_2G
+	select RAM_SET_1G
+	help
+	 This option configures 2G RAM if no luck then 1G
+
+config RAM_4G
+	bool "RAM_4G"
+	select RAM_SET_4G
+	help
+	 This option configures 4G RAM only
+
+config RAM_4_1G
+	bool "RAM_4_1G"
+	select RAM_SET_4G
+	select RAM_SET_1G
+	help
+	 This option configures 4G RAM if no luck then 1G
+
+config RAM_4_2_1G
+	bool "RAM_4_2_1G"
+	select RAM_SET_4G
+	select RAM_SET_2G
+	select RAM_SET_1G
+	help
+	 This option configures 4G RAM if no luck then 2G if no luck then 1G (does not work yet)
+
+endchoice
+
+config RAM_SET_4G
+	bool
+
+config RAM_SET_2G
+	bool
+
+config RAM_SET_1G
+	bool
+
+endif
diff --git a/configs/ucm-imx8_defconfig b/configs/ucm-imx8_defconfig
new file mode 100644
index 0000000..f8d6bd9
--- /dev/null
+++ b/configs/ucm-imx8_defconfig
@@ -0,0 +1,36 @@
+CONFIG_ARM=y
+CONFIG_ARCH_IMX8M=y
+CONFIG_SYS_MALLOC_F_LEN=0x2000
+CONFIG_TARGET_COMPULAB_IMX8=y
+CONFIG_TARGET_UCM_IMX8=y
+CONFIG_DEFAULT_DEVICE_TREE="ucm-imx8"
+CONFIG_LOCALVERSION="-ucm-imx8-1.0"
+CONFIG_FIT=y
+CONFIG_SPL_LOAD_FIT=y
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=arch/arm/imx-common/spl_sd.cfg"
+CONFIG_SPL=y
+CONFIG_HUSH_PARSER=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_CACHE=y
+CONFIG_CMD_PMIC=y
+CONFIG_CMD_REGULATOR=y
+CONFIG_CMD_FS_GENERIC=y
+CONFIG_OF_CONTROL=y
+CONFIG_OF_LIST="ucm-imx8"
+# CONFIG_BLK is not set
+CONFIG_DM_GPIO=y
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_MXC=y
+CONFIG_DM_MMC=y
+# CONFIG_DM_MMC_OPS is not set
+CONFIG_DM_ETH=y
+CONFIG_PINCTRL=y
+CONFIG_PINCTRL_IMX8M=y
+CONFIG_DM_PMIC=y
+CONFIG_DM_PMIC_PFUZE100=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_PFUZE100=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_DM_THERMAL=y
-- 
1.9.1

